test/test_macros.jl

```diff
@@ -1785,4 +1785,115 @@ function test_variable_Bool_argument()
     return
 end
 
+function test_nonlinear_generator_init_sum()
+    model = Model()
+    @variable(model, x)
+    a = @NLexpression(model, sum(x for i in 1:1; init = 0))
+    @test string(a) == "subexpression[1]: +x"
+    a = @NLexpression(model, sum(x for i in 1:0; init = 0))
+    @test string(a) == "subexpression[2]: 0.0"
+    a = @NLexpression(model, sum(x for i in 1:0; init = 1))
+    @test string(a) == "subexpression[3]: 1.0"
+    y = 3
+    a = @NLexpression(model, sum(x for i in 1:0; init = y))
+    @test string(a) == "subexpression[4]: 3.0"
+    a = @NLexpression(model, sum(x for i in 1:0, j in 1:0; init = 1))
+    @test string(a) == "subexpression[5]: 1.0"
+    a = @NLexpression(model, sum(x for i in 1:2, j in 1:2; init = 3))
+    @test string(a) == "subexpression[6]: x + x + x + x"
```

Comment:
So one issue here: it isn't obvious what to do with a non-empty expression and an `init`. The Julia docs say:
```julia
    sum(itr; [init])

  Returns the sum of all elements in a collection.

  The return type is Int for signed integers of less than system word size, and UInt for unsigned integers of less than system word size. For all other arguments, a common return type is found to which
  all arguments are promoted.

  The value returned for empty itr can be specified by init. It must be the additive identity (i.e. zero) as it is unspecified whether init is used for non-empty collections.
```
so it's undefined behavior.

At the REPL, it currently does:
```Julia
julia> sum(i for i in 1:3; init = 2)
8

julia> minimum(i for i in 1:3; init = 2)
1

julia> minimum(i for i in 1:3; init = -2)
-2
```

Reply:
cc @robbybp